ndcube/ndcube.py

```diff
@@ -918,6 +919,37 @@ def __neg__(self):
         return self._new_instance(data=-self.data)
 
     def __add__(self, value):
+        if isinstance(value, NDData) and value.wcs is None:
+            if self.unit is not None and value.unit is not None:
+                    value_data = value.data * value.unit.to(self.unit)
+            elif self.unit is None:
+                value_data = value.data
+            else:
+                raise TypeError("Cannot add unitless NDData to a unitful NDCube.")
+
+            # addition
+            new_data = self.data + value_data
+            # combine the uncertainty
+            new_uncertainty = None
+            if self.uncertainty is not None and value.uncertainty is not None:
+                new_uncertainty = self.uncertainty.propagate(
+                    np.add, value.uncertainty, correlation=0
+                )
+            elif self.uncertainty is not None:
+                new_uncertainty = self.uncertainty
+            elif value.uncertainty is not None:
+                new_uncertainty = value.uncertainty
+
+            # combine mask
+            self_ma = np.ma.MaskedArray(self.data, mask=self.mask)
+            value_ma = np.ma.MaskedArray(value_data, mask=value.mask)
+            result_ma = self_ma + value_ma
+            new_mask = result_ma.mask
+
+            # return the new NDCube instance
+            return self._new_instance(
+                data=new_data, uncertainty=new_uncertainty, mask=new_mask
+            )
```

Comment:
Instead of having a separate `return` here for the `NDData` case, I think we should build a dictionary of `kwargs` that we can give `self._new_instance`, [here](https://github.com/sunpy/ndcube/pull/794/files#diff-929794bb532be1486907b360285433b432f8505ab7e7ff63eca3eab657f7d00dR969).  So, you can create an empty kwargs dictionary at the start of the method, and add the new data, uncertainty, etc. in the relevant place, e.g.
```python
kwargs["uncertainty"] = new_uncertainty
```
Then the final line of the method would become
```python
return self._new_instance(**kwargs)
```

Reply:
Let me know if this doesn't make sense